ludwig/schema/encoders/image_encoders.py

```diff
@@ -284,368 +284,828 @@ class Stacked2DCNNEncoderConfig(BaseEncoderConfig):
     )
 
-    use_bias: Optional[bool] = schema_utils.Boolean(
+    saved_weights_in_checkpoint: bool = schema_utils.Boolean(
+        default=False,
+        description="Whether to save the weights in the checkpoint.",
+        parameter_metadata=ENCODER_METADATA["TVBaseEncoder"]["saved_weights_in_checkpoint"],
+    )
+
+    trainable: bool = schema_utils.Boolean(
         default=True,
-        description="Whether the layer uses a bias vector.",
-        parameter_metadata=ENCODER_METADATA["ResNetEncoder"]["use_bias"],
+        description="Is the encoder trainable.",
+        parameter_metadata=ENCODER_METADATA["TVBaseEncoder"]["trainable"],
     )
 
-    bias_initializer: Optional[str] = schema_utils.StringOptions(
-        sorted(list(initializer_registry.keys())),
-        default="zeros",
-        description="initializer for the bias vector.",
-        parameter_metadata=ENCODER_METADATA["ResNetEncoder"]["bias_initializer"],
+
+@register_encoder_config("alexnet_torch", IMAGE)
+@dataclass
+class TVAlexNetEncoderConfig(TVBaseEncoderConfig):
+    type: str = schema_utils.StringOptions(
+        ["alexnet_torch"],
+        default="alexnet_torch",
+        allow_none=False,
+        description="Type of encoder.",
     )
 
-    weights_initializer: Optional[str] = schema_utils.StringOptions(
-        sorted(list(initializer_registry.keys())),
-        default="xavier_uniform",
-        description="Initializer for the weights matrix.",
-        parameter_metadata=ENCODER_METADATA["ResNetEncoder"]["weights_initializer"],
+    model_variant: Optional[str] = schema_utils.StringOptions(
+        ["base"],
+        default="base",
+        allow_none=False,
+        description="Pretrained model variant to use.",
+        parameter_metadata=ENCODER_METADATA["TVAlexNetEncoder"]["model_variant"],
     )
 
-    output_size: Optional[int] = schema_utils.PositiveInteger(
-        default=128,
-        description="if output_size is not already specified in fc_layers this is the default output_size that will "
-        "be used for each layer. It indicates the size of the output of a fully connected layer. ",
-        parameter_metadata=ENCODER_METADATA["ResNetEncoder"]["output_size"],
+
+@register_encoder_config("convnext_torch", IMAGE)
+@dataclass
+class TVConvNeXtEncoderConfig(TVBaseEncoderConfig):
+    type: str = schema_utils.StringOptions(
+        ["convnext_torch"],
+        default="convnext_torch",
+        allow_none=False,
+        description="Type of encoder.",
     )
 
-    norm: Optional[str] = schema_utils.StringOptions(
-        ["batch", "layer"],
-        default=None,
-        description="if a norm is not already specified in fc_layers this is the default norm that will be used for "
-        "each layer. It indicates the norm of the output and can be null, batch or layer.",
-        parameter_metadata=ENCODER_METADATA["ResNetEncoder"]["norm"],
+    model_variant: Optional[str] = schema_utils.StringOptions(
```

Comment:
Want to double-check if this field is really `Optional`? If so, make sure `allow_none` is `True`.

Reply:
done